app/views/contexts/summary/list_collector_base_block.py

```diff
@@ -0,0 +1,200 @@
+from app.views.contexts import list_context
+from app.views.contexts.summary.block import Block
+
+
+class ListCollectorBaseBlock:
+    def __init__(
+        self,
+        *,
+        routing_path_block_ids: Iterable[str],
+        answer_store: AnswerStore,
+        list_store: ListStore,
+        progress_store: ProgressStore,
+        metadata: MetadataProxy | None,
+        response_metadata: MutableMapping,
+        schema: QuestionnaireSchema,
+        location: Location,
+        language: str,
+        supplementary_data_store: SupplementaryDataStore,
+        return_to: str | None,
+        return_to_block_id: str | None = None,
+    ) -> None:
+        self._location = location
+        self._placeholder_renderer = PlaceholderRenderer(
+            language=language,
+            answer_store=answer_store,
+            list_store=list_store,
+            metadata=metadata,
+            response_metadata=response_metadata,
+            schema=schema,
+            progress_store=progress_store,
+            location=location,
+            supplementary_data_store=supplementary_data_store,
+        )
+        self._list_store = list_store
+        self._schema = schema
+        self._location = location
+        # type ignore added as section should exist
+        self._section: ImmutableDict = self._schema.get_section(self._location.section_id)  # type: ignore
+        self._language = language
+        self._answer_store = answer_store
+        self._metadata = metadata
+        self._response_metadata = response_metadata
+        self._routing_path_block_ids = routing_path_block_ids
+        self._progress_store = progress_store
+        self._supplementary_data_store = supplementary_data_store
+        self._return_to = return_to
+        self._return_to_block_id = return_to_block_id
+
+    @property
+    def list_context(self) -> list_context.ListContext:
+        return list_context.ListContext(
+            self._language,
+            self._schema,
+            self._answer_store,
+            self._list_store,
+            self._progress_store,
+            self._metadata,
+            self._response_metadata,
+            self._supplementary_data_store,
+        )
+
+    def _list_collector_block_on_path(self, for_list: str) -> list:
+        list_collector_blocks = list(
+            self._schema.get_list_collectors_for_list(self._section, for_list=for_list)
+        )
+
+        return [
+            list_collector_block
+            for list_collector_block in list_collector_blocks
+            if list_collector_block["id"] in self._routing_path_block_ids
+        ]
+
+    def _list_collector_block(
+        self, for_list: str, list_collector_blocks_on_path: list
+    ) -> ImmutableDict:
+        list_collector_blocks = list(
+            self._schema.get_list_collectors_for_list(self._section, for_list=for_list)
+        )
+        # Type ignore: self._schema.get_list_collectors_for_list can return None
```

Comment:
Appreciate this was already there, but the type ignore has an incorrect comment, `list_collector_blocks` will be a list of Immutable dictionaries and the reason for the type ignore is that `list_collector_blocks_on_path` is of type `list` - so if the type of that is just corrected this ignore can be removed